### PR TITLE
Test command

### PR DESCRIPTION
**Pull request by exekias**
This PR adds a self-testing framework, different parts of beats can implement `Testable` interface to allow the user to test her settings.
Current features:
* Test utils with output to console
* Output testing (ES & Logstash)

example ERROR:
```
X1 :: elastic/beats/filebeat ‹test-command*› » ./filebeat test output                                              1 ↵
elasticsearch: https://badhost:9243...
  parse url... OK
  connection...
    parse host... OK
    dns lookup... ERROR lookup badhost: no such host
```

example OK:
```
X1 :: elastic/beats/filebeat ‹test-command*› » ./filebeat test output 
elasticsearch: https://61fc3e1983ebd7b13e9d6098e4651345.us-east-1.aws.found.io:9243...
  parse url... OK
  connection...
    parse host... OK
    dns lookup... OK
    addresses: 54.235.139.222, 54.221.244.80, 54.243.122.128, 54.204.28.14, 54.221.223.25, 54.243.118.44, 54.243.142.98, 54.235.122.205
    dial up... OK
  TLS...
    security: server's certificate chain verification is enabled
    handshake... OK
    TLS version: TLSv1.2
    dial up... OK
  talk to server... OK
  version: 5.5.0
```
TODO:
- [x] Unit tests
- [x] Integration tests

### CODE REVIEW

File: libbeat/tests/system/test_cmd.py [hunk 1]
```diff
@@ -132,5 +133,40 @@ def test_export_template(self):
...
+                                    os.path.join(self.working_dir,
+                                                 "mockbeat.yml"),
+                                    elasticsearch={"hosts": '["badhost:9200"]'})
+        exit_code = self.run_beat(
+            extra_args=["test", "output"],
+            config="mockbeat.yml")
+
+        assert exit_code == 1
+        assert self.log_contains('parse url... OK')
+        assert self.log_contains('dns lookup... ERROR lookup badhost: no such host')
```
Comment (andrewkroh) [inline libbeat/tests/system/test_cmd.py]: I would be careful with matching the specific error message coming from the stdlib or the OS. This has bit me a few times when running my tests on different OSes and Golang versions.

```
elasticsearch: http://badhost:9200...
  parse url... OK
  connection...
    parse host... OK
    dns lookup... ERROR lookup badhost on 127.0.0.11:53: no such host
```

```
======================================================================
FAIL: Test test wrong output works
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/go/src/github.com/elastic/beats/libbeat/tests/system/test_cmd.py", line 169, in test_test_wrong_output
    assert self.log_contains('dns lookup... ERROR lookup badhost: no such host')
```

**Comment by exekias** [comment]
jenkins retest this please

**Comment by exekias** [comment]
thanks for the review, btw I had a pending check that color works on Mac, I guess you just confirmed it?

**Comment by andrewkroh** [comment]
Yeah, the color works in Terminal on macOS.

**Pull request merged**